core/domain/exp_services.py

```diff
@@ -1477,10 +1456,8 @@ def get_demo_exploration_components(demo_path):
     if demo_filepath.endswith('yaml'):
         file_contents = utils.get_file_contents(demo_filepath)
         return file_contents, []
-    elif os.path.isdir(demo_filepath):
-        return utils.get_exploration_components_from_dir(demo_filepath)
     else:
-        raise Exception('Unrecognized file path: %s' % demo_path)
```

Comment:
Why are we deleting this?

Reply:
Ah I just shifted this check to its test class as its not getting called here

Reply:
Have we made the line 1480 check anywhere before calling `get_exploration_components_from_dir`?

> 1480     elif os.path.isdir(demo_filepath):

Reply:
Why this condition would be necessary now? If "if" fails, the other line's gonna execute and the test `test_valid_demo_file_path` is gonna check for valid demo file paths

Reply:
> Have we made the line 1480 check anywhere before calling get_exploration_components_from_dir?
> > Why this condition would be necessary now?

Just to make sure that we are calling `get_exploration_components_from_dir` with a correct dir_path.

> If "if" fails, the other line's gonna execute and the test test_valid_demo_file_path is gonna check for valid demo file paths

Sorry, I'm unable to understand this! :(

Reply:
In short yes we are checking for the condition(`elif os.path.isdir(demo_filepath):`) in the test function test_valid_demo_file_path.

Reply:
Why we are adding such logic/function in the tests? I think the test files should only test the behavior of the main functionality instead of creating a test functionality which isn't a part of the main method.

CAn you share the link of the method where you have implemented `test_valid_demo_file_path`?

Reply:
The exception will be raised if the path is invalid and that will happen if the path in feconf changes and the test file prevents this to happen.

Reply:
>  if the path in feconf changes and the test file prevents this to happen.

That's a good test, thanks for adding it. :)

but the changes in the method which is calling `get_demo_exploration_components` can cause this issue as well, let's prevent this to happen here.

Reply:
As far as I can see, load_demo is the only function that calls get_demo_exploration_components but still as I mentioned the exception will be raised only when the path in feconf changes and we are not changing it in load_demo. Can you please explain how we can raise the exception in this function?

Reply:
Suppose someone changed the `load_demo` function such that it sends a path to `get_demo_exploration_components` which isn't a dir (or yaml file). Which test case will catch this?

Reply:
Reverted the change and added tests.